stock_picking_batch_creation/wizards/make_picking_batch.py

```diff
@@ -0,0 +1,404 @@
+        required=True,
+    )
+    group_pickings_by_partner = fields.Boolean(
+        default=False,
+        string="Group pickings by partner",
+        help="All the pickings related to one partner will be put in one bin",
+    )
+    picking_locking_mode = fields.Selection(
+        selection=[
+            ("sql_for_update_skip_locked", "SQL FOR UPDATE SKIP LOCKED"),
+        ],
+        default=None,
+        string="Picking locking mode",
+        help="Define the way the system will search and lock the pickings. "
+        "In the same time, picking already locked by another transaction will "
+        "be skipped. This should reduce the risk of deadlocks if 2 users "
+        "try to create a batch at the same time.",
+    )
+    add_picking_list_in_error = fields.Boolean(
+        default=False,
+        string="Add all the names of the pickings in error message",
+        help="If not suitable device is provided for the pickings candidates, "
+        "the error message will contain the list of the pickings names. In some"
+        "cases, this list can be very long. That's why this option is unchecked"
+        "by default.",
+    )
+
+    __slots__ = (
+        "_volume_by_partners",
+        "_device",
+        "_remaining_weight",
+        "_remaining_nbr_picking_lines",
+        "_selected_picking_ids",
+        "_first_picking",
+        "_remaining_nbr_bins",
+        "_previous_selected_picking",
+    )
+
+    def __init__(self, env, ids=(), prefetch_ids=()):
+        super().__init__(env, ids=ids, prefetch_ids=prefetch_ids)
+        self._reset_counters()
+
+    def create_batch(self):
+        self.ensure_one()
+        batch = self._create_batch(raise_if_not_possible=True)
+        action = {
+            "type": "ir.actions.act_window",
+            "name": batch.name,
+            "res_model": "stock.picking.batch",
+            "res_id": batch.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+        return action
+
+    def _reset_counters(self):
+        self._volume_by_partners = defaultdict(lambda: 0)
+        self._device = None
+        self._remaining_weight = 0
+        self._remaining_nbr_picking_lines = 0
+        self._selected_picking_ids = []
+        self._first_picking = None
+        self._remaining_nbr_bins = None
+        self._previous_selected_picking = None
+
+    def _get_picking_order_by(self):
+        self.ensure_one()
+        if self.group_pickings_by_partner:
+            return "priority desc, scheduled_date asc, partner_id desc, id asc"
+        return "priority desc, scheduled_date asc, id asc"
+
+    def _get_picking_domain_common(self, user=None):
+        domain = [
+            ("picking_type_id", "in", self.picking_type_ids.ids),
+            ("state", "in", ("partially_available", "assigned")),
+            ("batch_id", "=", False),
+            ("user_id", "=", user.id if user else False),
```

Comment:
In all shopfloor scenario, what is assigned to a responsible/user can be processed by the user only, what is not assigned can be processed by anybody
```suggestion
            ("user_id", "in", [user.id, False] if user else [False]),
```

Reply:
Ok, I see, the call is done twice, once with the user and once without. If you do the change above and add user_id in the sorting, I think you should be able to drop the second call